lib/get-repositories.js

```diff
@@ -1,9 +1,24 @@
 const fs = require('fs');
+const { isArray } = require('lodash');
 const { GITHUB_REPO_REGEX, extractOwnerAndRepo } = require('./github-helpers');
 
-const getRepositories = limit => {
-	const input = fs.readFileSync('/dev/stdin') || '';
-	const inputRepositories = input.toString().split('\n');
+const getRepositories = ({ limit, repoList } = {}) => {
+	let inputRepositories;
+
+	const hasStdIn = !process.stdin.isTTY;
+
+	if (isArray(repoList) && !!repoList.length && hasStdIn) {
+		throw new Error(
+			'choose either to pipe through a repo list OR pass it as args'
+		);
+	}
+
+	if (hasStdIn) {
+		const input = fs.readFileSync('/dev/stdin');
```

Comment:
`fs.readFileSync` could be converted into a function that returns a promise, then you could `await` on the result. e.g.

```js
const read = utils.promisfy(fs.readFile);
const input = await read('/dev/stdin');
```

Not necessary to change this, only an FYI.

Reply:
http://2ality.com/2017/05/util-promisify.html

Reply:
Thanks - very interesting.